src/object_store/src/object/s3.rs

```diff
@@ -716,11 +716,28 @@ impl S3ObjectStore {
             .send()
             .await;
         let mut is_expiration_configured = false;
+
         if let Ok(config) = &get_config_result {
             for rule in config.rules().unwrap_or_default() {
-                if rule.expiration().is_some() {
-                    is_expiration_configured = true;
+                // Check if the filter is not set or the prifix in the filter is data directory in RisingWave,
+                // and if the expiration status of the rule is "Enabled".
+                // If both conditions are met, it is considered that there is a risk of data deletion.
+                match rule.filter().as_ref() {
+                    Some(&LifecycleRuleFilter::Prefix(prefix)) => {
+                        if let Some(ExpirationStatus::Enabled) = rule.status && prefix == data_directory{
```

Comment:
better to check whether data directory contains prefix.

Reply:
+1

Reply:
Nice catch.